hello-world/src/lib.rs

```diff
@@ -6,74 +6,109 @@ static mut TAMAGOTCHI: Option<Tamagotchi> = None;
 
 #[derive(Default, Encode, Decode, TypeInfo)]
 pub struct Tamagotchi {
-   pub owner: ActorId,
    pub name: String,
    pub date_of_birth: u64,
+   pub owner: ActorId,
 
-   pub fed: u16,
-   pub happy: u16,
-   pub rested: u16,
+   pub fed: u64,
+   pub fed_block: u64,
+   pub entertained: u64,
+   pub entertained_block: u64,
+   pub rested: u64,
+   pub rested_block: u64,
 
-   pub last_ate: u32,
-   pub last_had_fun: u32,
-   pub last_slept: u32
+   pub allowed_account: Option<ActorId>,
 }
 
 impl Tamagotchi {
+    fn transfer(&mut self, actor_id: ActorId) {
+        let sender = msg::source();
+        assert!(
+            sender == self.owner || sender == self.allowed_account.unwrap_or_default(),
```

Comment:
`unwrap_or_default` is not appropriate here. Better to compare `Option`s:
```suggestion
            sender == self.owner || self.allowed_account == Some(sender),
```